x-pack/plugins/spaces/server/lib/saved_objects_client/get_namespace.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { BaseOptions, SavedObjectsService } from 'src/legacy/server/saved_objects';
+import { DEFAULT_SPACE_ID } from '../../../common/constants';
+
+export const createGetNamespace = (savedObjectsService: SavedObjectsService) => {
+  return function getNamespace(operationOptions: BaseOptions, currentSpaceId: string) {
+    if (operationOptions.namespace) {
+      return operationOptions.namespace;
+    }
+    if (currentSpaceId === DEFAULT_SPACE_ID) {
+      return undefined;
```

Comment:
It feels like we should be using `savedObjectsService.createNamespace(undefined)` here, since we're explicitly stating that we should be looking for objects without a namespace and this is what consumers of the fully wrapped SOC would be using to represent the default namespace.

Reply:
Yep, good call! I'll make this change